### PR TITLE
Kaggle learning

### PR DESCRIPTION
**Pull request by CecileSerene**
Compare 5 classifiers:
* Logistic Regression
* KNN
* Linear SVM
* Random Forest
* Naive Bayes

Capture d’écran 2019-03-31 à 14 02 32

Still need to add this to a jupyter notebook

### CODE REVIEW

**Comment by dlphn** [comment]
À merger sur kaggle_dataset non ?

**Comment by florimondmanca** [comment]
@dlphn Si Cécile est partie de `master` pour faire cette branche je dirais que non. De toute façon il faudrait merge `kaggle_dataset` asap non ? Parce que c'est un peu le bazar. 😅

**Comment by florimondmanca** [comment]
Je confirme @dlphn je crois que la base c'était `kaggle_dataset` du coup je viens de la changer.

**Review by florimondmanca — APPROVED** [review]

File: classifier_comparison.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import time
+import warnings
+from IPython.display import display
+
+from preprocessing import format_text
+
+from sklearn.pipeline import Pipeline
```
Comment (dlphn) [inline classifier_comparison.py:12]: Enlever les imports non-utilisés

File: classifier_comparison.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        dict_models[classifier_name]['f_score'] = f_score
+
+    def compare(dataset):
+        """
+        Evaluate each model in turn
+        """
+        results = []
+        names = []
+        # https://scikit-learn.org/stable/modules/model_evaluation.html
+        scoring = 'accuracy'
```
Comment (dlphn) [inline classifier_comparison.py:165]: Essayer une autre mesure car l'accuracy est biaisée vu que nos données sont disproprotionnées

**Review by dlphn — APPROVED** [review]
C'est assez long à run de mon côté, est-ce que c'est pareil pour vous ? Mais sinon ça fonctionne bien

**Comment by CecileSerene** [comment]
> C'est assez long à run de mon côté, est-ce que c'est pareil pour vous ? Mais sinon ça fonctionne bien

Oui pour moi aussi c'est très long, je sais pas trop pourquoi, pourtant je fais bien attention à ne vectoriser le texte que une seule fois...
